packages/metadataform-extension/src/index.ts

```diff
@@ -225,27 +221,20 @@ namespace Private {
           }
 
           // Optionally links key to a custom widget.
-          if (options.customWidget) {
-            const formWidget = formWidgetsRegistry.getRenderer(
-              options.customWidget as string
+          if (options.customComponent) {
+            const component = formComponentRegistry.getComponent(
+              options.customComponent as string
             );
 
             // If renderer is defined (custom widget has been registered), set it as used widget.
-            if (formWidget !== undefined)
+            if (component !== undefined) {
               if (!uiSchema[metadataKey]) uiSchema[metadataKey] = {};
-            uiSchema[metadataKey]['ui:widget'] = formWidget;
-          }
-
-          // Optionally links key to a custom field.
-          if (options.customField) {
-            const formField = formComponentRegistry.getRenderer(
-              options.customField as string
-            );
-
-            // If renderer is defined (custom widget has been registered), set it as used widget.
-            if (formField !== undefined)
-              if (!uiSchema[metadataKey]) uiSchema[metadataKey] = {};
-            uiSchema[metadataKey]['ui:field'] = formField;
+              if (component.fieldRenderer) {
+                uiSchema[metadataKey]['ui:field'] = component.fieldRenderer;
+              } else {
+                uiSchema[metadataKey]['ui:widget'] = component.widgetRenderer;
+              }
```

Comment:
Should we support (and does it makes sense) to have simultaneously `ui:field` and `ui:widget` defined?

Reply:
I may be wrong but I don't think it make sense.
The `ui:widget` is used to replace the default widget for an input.
If a custom field is defined, there is no default widget anymore, as the field should contain one or more widgets.